### PR TITLE
PLT-637 Search logs

### PR DESCRIPTION
**Pull request by nikhilbonte21**

### CODE REVIEW

File: addons/elasticsearch/es-search-logs-mappings.json [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        "type": "text"
+      },
+      "errorCode": {
+        "type": "keyword"
+      },
+
+      "resultsCount": {
+        "type": "long"
+      },
+      "timestamp": {
```
Comment (n5nk) [inline addons/elasticsearch/es-search-logs-mappings.json:64]: lets be specific here, what does `timestamp` represent ? is this creationTime or something else.

File: addons/elasticsearch/es-search-logs-mappings.json [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        "type": "keyword"
+      },
+      "utmTags": {
+        "type": "keyword"
+      },
+
+      "hasResult": {
+        "type": "boolean"
+      },
+      "isFailed": {
```
Comment (n5nk) [inline addons/elasticsearch/es-search-logs-mappings.json:50]: `isFailed` represents what ? Does this infer request failed ?

File: authorization/src/main/java/org/apache/atlas/authorize/simple/AtlasSimpleAuthorizer.java [hunk 3]
```diff
@@ -350,6 +350,11 @@ public void scrubSearchResults(AtlasSearchResultScrubRequest request) throws Atl
         }
     }
 
+    @Override
+    public void scrubSearchResults(AtlasSearchResultScrubRequest request, boolean isScrubAuditEnabled) throws AtlasAuthorizationException {
+        scrubSearchResults(request);
```
Comment (n5nk) [inline authorization/src/main/java/org/apache/atlas/authorize/simple/AtlasSimpleAuthorizer.java:355]: variable `isScrubAuditEnabled` not used

File: addons/elasticsearch/es-search-logs-mappings.json [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      "ipAddress": {
+        "type": "keyword"
+      },
+      "userName": {
+        "type": "keyword"
+      },
+
+
+      "entityGuidsAll": {
+        "type": "keyword"
```
[the remark this answers is not in the document]
Comment (nikhilbonte21) [reply addons/elasticsearch/es-search-logs-mappings.json:26]: yes, it is list of String

File: repository/src/main/java/org/apache/atlas/discovery/EntityDiscoveryService.java [hunk 5]
```diff
@@ -1004,6 +1008,31 @@ public AtlasSearchResult directIndexSearch(SearchParams searchParams) throws Atl
...
+    public SearchLogSearchResult searchLogs(SearchLogSearchParams searchParams) throws AtlasBaseException {
+        SearchLogSearchResult ret = new SearchLogSearchResult();
+        ret.setSearchParameters(searchParams);
+        AtlasIndexQuery indexQuery = null;
+
+        try {
+            indexQuery = graph.elasticsearchQuery(ESSearchLogger.INDEX_NAME);
+            Map<String, Object> result = indexQuery.directIndexQuery(searchParams.getQueryString());
+
+            ret.setApproximateCount( ((Integer) result.get("total")).longValue());
```
[the remark this answers is not in the document]
Comment (nikhilbonte21) [reply repository/src/main/java/org/apache/atlas/discovery/EntityDiscoveryService.java:1021]: added a null check

File: addons/elasticsearch/es-search-logs-mappings.json [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+      },
+
+      "resultsCount": {
+        "type": "long"
+      },
+      "timestamp": {
+        "type": "long"
+      },
+      "created": {
```
[the remark this answers is not in the document]
Comment (nikhilbonte21) [reply addons/elasticsearch/es-search-logs-mappings.json:67]: it is createdAt, making necessary changes

File: addons/elasticsearch/es-search-logs-mappings.json (hunk 2 above)
Comment (nikhilbonte21) [reply addons/elasticsearch/es-search-logs-mappings.json:50]: Yes, it tells whether indexsearch request was failed or successful

File: authorization/src/main/java/org/apache/atlas/authorize/simple/AtlasSimpleAuthorizer.java (hunk 3 above)
Comment (nikhilbonte21) [reply authorization/src/main/java/org/apache/atlas/authorize/simple/AtlasSimpleAuthorizer.java:355]: We will handle this in [this](https://linear.app/atlanproduct/issue/PLT-636/return-personapurposepolicy-details-in-the-scrubbed-assets) chnage

File: intg/src/main/java/org/apache/atlas/model/searchlog/SearchRequestLogData.java [hunk 7]
```diff
@@ -0,0 +1,325 @@
...
+        private boolean hasResult;
+        private boolean isFailed;
+
+        private long resultsCount;
+        private long responseTime;
+        private long timestamp;
+
+        public SearchRequestLogDataBuilder(){}
+
+        /*public SearchRequestLogDataBuilder setDsl(Map<String, Object> dsl) {
```
[the remark this answers is not in the document]
Comment (nikhilbonte21) [reply intg/src/main/java/org/apache/atlas/model/searchlog/SearchRequestLogData.java:213]: We are focussing [on v1](https://www.notion.so/atlanhq/Search-Logs-101-1c559d7a12614875bbebd71f0399cd42?pvs=4#8e382c6ee3b042de94a03dbc5972c5c8) with this rollout. We would have kept it but this actually causes issue of mapping limit exceeded

This happens due to we marked it as Nested in ES & for each key in DSL, a new mapping is being created

File: graphdb/janus/src/main/java/org/apache/atlas/repository/graphdb/janus/AtlasJanusIndexQuery.java [hunk 8]
```diff
@@ -50,6 +51,11 @@ public DirectIndexQueryResult<AtlasJanusVertex, AtlasJanusEdge> vertices(SearchP
         return null;
     }
 
+    @Override
```
[the remark this answers is not in the document]
Comment (nikhilbonte21) [reply graphdb/janus/src/main/java/org/apache/atlas/repository/graphdb/janus/AtlasJanusIndexQuery.java:54]: Object of this class is being initialised at https://github.com/atlanhq/atlas-metastore/blob/5533cb700cceda469b8acd2a0848953b3e8e6c3a/graphdb/janus/src/main/java/org/apache/atlas/repository/graphdb/janus/AtlasJanusGraph.java#L310, can not mark it as abstract

File: addons/elasticsearch/es-search-logs-mappings.json (hunk 1 above)
Comment (nikhilbonte21) [reply addons/elasticsearch/es-search-logs-mappings.json:64]: This is creation time, will keep it same to be in sync with janusgraph_vertex_index's __timestamp

**Review by n5nk — APPROVED** [review]
LGTM

**Pull request merged**